metric-anomaly-data-model/src/main/java/org/hypertrace/alert/engine/metric/anomaly/datamodel/queue/KafkaConfigReader.java

```diff
@@ -33,22 +48,55 @@ public String getBootstrapServer() {
     return mergeProperties(baseProps, getFlatMapConfig(kafkaQueueConfig, CONSUMER_CONFIG));
   }
 
+  public Map<String, Object> getProducerSerdeProperties() {
+    String schemaRegistryUrl =
+        new StringJoiner(".").add(CONSUMER_CONFIG).add(SCHEMA_REGISTRY_URL).toString();
+    Map<String, Object> map = new HashMap<>();
+    if (kafkaQueueConfig.hasPath(schemaRegistryUrl)) {
```

Comment:
we can put all these additional properties in application.conf file in PRODUCER_CONFIG section eg. `schema.registry.url`, and serde properties
we just have to put the default value in the corresponding class's base properties. So, would suggest, add below to base properties:

```
ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, AvroSerializer.class.getName()

ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, AvroSerializer.class.getName()
```
And, update the application.conf file with 
```
schema.registry.url = 
key.serializer= 
value.serializer
```
The method `getProducerConfig` merges application.conf properties with baseProperties

Reply:
Makes sense, updated